### DISCUSSION TITLE
Disclouser animation

### DISCUSSION THREAD

**Discussion by mule-stand** [ariakit/ariakit#3495] [Questions] [answered]
How add CSS animation to Disclouser? `animated` prop seems to work but it is deprecated. What are the best solutions now?
https://ariakit.org/reference/disclosure

**Comment by diegohaz** [comment, marked as answer]
You must apply CSS transition or animation to the disclosure content element. The same way you would do with a [dialog](https://ariakit.org/examples/dialog-animated) (which is built upon disclosure).

**Reply by mule-stand** [reply]
Do I always have to use alwaysVisible when using framer motion?

**Reply by diegohaz** [reply]
Yes

**Reply by mule-stand** [reply]
Doesn't `unmountOnHide` work the same as `mounted &&` (`store.useState("mounted")`)?

**Reply by diegohaz** [reply]
Not exactly. `unmountOnHide` will still mount the `<Dialog>` component, but it won't mount the inner component where the entire dialog logic resides. On the other hand, `mounted && <Dialog />` unmounts the `Dialog` component itself, which is the direct child of `AnimatePresence`. To the best of my knowledge, `AnimatePresence` requires its direct child to be unmounted, which is why it works with `mounted &&`, but not with `unmountOnHide`.

**Reply by mule-stand** [reply]
And this if I'm not mistaken can affect accessibility, because some elements should always be in the DOM for accessibility and their correct processing by screen readers?

**Reply by diegohaz** [reply]
It depends. What's the concrete use case?

**Reply by mule-stand** [reply]
For example, with the same `Dialog`, `Disclouser`, or `FormError`
